drivers/regulator/regulator_fixed.c

```diff
@@ -19,11 +19,9 @@ LOG_MODULE_REGISTER(regulator_fixed, CONFIG_REGULATOR_LOG_LEVEL);
 
 struct driver_config {
 	const char *regulator_name;
-	const char *gpio_name;
 	uint32_t startup_delay_us;
 	uint32_t off_on_delay_us;
-	gpio_pin_t gpio_pin;
-	gpio_dt_flags_t gpio_flags;
+	struct gpio_dt_spec enable;
```

Comment:
Please rename to `struct gpio_dt_spec gpio;` or `struct gpio_dt_spec reg_gpio;`

Reply:
Why is that? The devicetree property is called `enable-gpios`

Reply:
Maybe `enable_pin` or `enable_gpio`.

Reply:
Looking through the existing usage of `struct gpio_dt_spec`, it looks like a pretty even split between including `gpio` in the name and not, with an edge towards not (influenced by my merged PR's that don't #37442, #37514)

My position is that these variables are only used in functions where it is exceedingly obvious that the object is a gpio (`gpio_configure_dt()`, etc), so the added value of tacking on `_gpio` isn't that high. Types aren't duplicated in the names of other fields, why is it needed here?

Reply:
> Why is that? The devicetree property is called `enable-gpios`

That should probably just be 'gpio'.

> My position is that these variables are only used in functions where it is exceedingly obvious that the object is a gpio (`gpio_configure_dt()`, etc), so the added value of tacking on `_gpio` isn't that high. Types aren't duplicated in the names of other fields, why is it needed here?

I am also fine with: onoff, offon, or disable.

Reply:
> That should probably just be 'gpio'.

That is not valid for a GPIO property, they must be postfixed with `-gpios`. See https://docs.zephyrproject.org/latest/guides/dts/bindings.html#specifier-cell-names-cells

> I am also fine with: onoff, offon, or disable.

Semantically, setting `onoff` to `true` or `false`, as in `gpio_pin_set_dt()`, is not entirely clear, does `true` refer to the on or the off part? Same problem with `offon`. `disable` implies the exact opposite of what is being done, `gpio_pin_set_dt(disable, true)` is turning the regulator on?

Reply:
>gpio_pin_set_dt(disable, true) is turning the regulator on?

It is confusing, is not it? So what is the problem with reg_gpio (or just reg) ?

Reply:
> It is confusing, is not it?

Yes, because control pins on regulators are called the output enable pin, not the output disable pin, and comparing the two:
```
gpio_pin_set_dt(&cfg->enable, true);
gpio_pin_set_dt(&cfg->disable, false); // Double negative to turn the device on
```

> So what is the problem with reg_gpio (or just reg) ?

Because every regulator I have seen calls the control pin Output Enable (OE). Why use some new term to refer to something that is already standardized. I don't believe anyone looking at a regulator driver and seeing an `enable` pin will be confused, but `reg` has no meaning beyond the fact that the device is a regulator.

Reply:
> Because every regulator I have seen calls the control pin Output Enable (OE)

Indeed, good point, please rename it to `oe` then, thanks.

Reply:
Absent other considerations, I feel that an argument could be made for `output_enable` vs `enable`. The `oe` abbreviation is IMO too abbreviated.

I have chosen `enable` to match the devicetree specification. If you believe that `enable` is a poor term to describe the pin, that can be addressed in another PR.